.github/workflows/nativeAOT.yml

```diff
@@ -0,0 +1,90 @@
+name: NativeAOT
+
+on:
+  push:
+    branches:
+      - main
+      - 'hotfix/**'
+    tags:
+      - '*'
+  pull_request:
+
+env:
+  dotnet_sdk_version: '7.0.102'
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-22.04]
+        pg_major: [15]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: NuGet Cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.nuget/packages
+          key: ${{ runner.os }}-nuget-${{ hashFiles('**/Directory.Build.targets') }}
+          restore-keys: |
+            ${{ runner.os }}-nuget-
+
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v3.0.3
+        with:
+          dotnet-version: |
+            ${{ env.dotnet_sdk_version }}
+
+      - name: Setup Native AOT prerequisites
+        run: sudo apt-get install clang zlib1g-dev
+        shell: bash
+
+      - name: Build
+        run: dotnet publish test/Npgsql.NativeAotTests/Npgsql.NativeAotTests.csproj -r linux-x64 -c Release -f net7.0
+        shell: bash
+
+      - name: Start PostgreSQL ${{ matrix.pg_major }} (Linux)
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          # First uninstall any PostgreSQL installed on the image
+          dpkg-query -W --showformat='${Package}\n' 'postgresql-*' | xargs sudo dpkg -P postgresql
+
+          # Import the repository signing key
+          wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
+
+          sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ jammy-pgdg main ${{ matrix.pg_major }}" >> /etc/apt/sources.list.d/pgdg.list'
+          sudo apt-get update -qq
+          sudo apt-get install -qq postgresql-${{ matrix.pg_major }}
+          export PGDATA=/etc/postgresql/${{ matrix.pg_major }}/main
+
+          sudo -u postgres psql -c "CREATE USER npgsql_tests SUPERUSER PASSWORD 'npgsql_tests'"
+          sudo -u postgres psql -c "CREATE DATABASE npgsql_tests"
+
+          sudo pg_ctlcluster ${{ matrix.pg_major }} main restart
+
+      # Uncomment the following to SSH into the agent running the build (https://github.com/mxschmitt/action-tmate)
+      #- uses: actions/checkout@v3
+      #- name: Setup tmate session
+      #  uses: mxschmitt/action-tmate@v3
+
+      - name: Run
+        run: test/Npgsql.NativeAotTests/bin/Release/net7.0/linux-x64/native/Npgsql.NativeAotTests
+
+      - name: Dump mstat
```

Comment:
I think there should be a way to make this actually appear in the CI runs in the build summary (like that test summary we have in our runs) - in case you feel like going into that.

Reply:
Done!